### DISCUSSION TITLE
SPI.cpp sets _ss but never uses it.

### DISCUSSION THREAD

**Discussion by stolk** [espressif/arduino-esp32#6587] [General]
I don't understand why [SPI.cpp](https://github.com/espressif/arduino-esp32/blob/master/libraries/SPI/src/SPI.cpp) sets a member variable `_ss` but then never uses it?

What is the fuction of the `_ss` variable in this class?

I take it that `_ss` stands for slave select?

Why is it associated with the SPI bus?

Is the function of slave select the same as chip select? If so, it should not be associated with the SPI bus?

**Comment by me-no-dev** [comment]
The SPI bus can manage the SS pin by itself, though we do not usually do this in Arduino. The option is there for the people that might need it.
